runtime/src/main/cpp/utf8/with_replacement.h

```diff
@@ -37,13 +37,13 @@ uint32_t next(octet_iterator &it, const octet_iterator end, uint32_t replacement
         case internal::UTF8_OK :
             return cp;
         case internal::INVALID_LEAD :
+        case internal::INVALID_CODE_POINT :
```

Comment:
Why this change?

Reply:
All modifications in this(`with_replacement.h`) and `core.h` files are to change the logic of replacing invalid code points.
The idea is to have identical replacement strategy on all (where feasible) Kotlin platforms, following unicode standard recommendations on handling invalid sequences.
Having said that, unicode standard recommends replacing each byte of an invalid code point with a single replacement code point.

Reply:
Just to be clear, it's not a _recommendation_ of the Unicode standard, but rather a requirement of the W3C standard. The Unicode standard now describes it as a practice of "An increasing number of implementations".
See the section "U+FFFD Substitution of Maximal Subparts" here: http://www.unicode.org/versions/Unicode12.1.0/ch03.pdf#page=57